File: java/src/main/java/org/example/domain/Match.java
package org.example.domain;

public class Match extends Entity<Long> {
    private String name;
    private double ticketPrice;
    private int totalSeats;
    private int availableSeats;

    public Match() {}

    public Match(Long id, String name, double ticketPrice, int totalSeats, int availableSeats) {
        super(id);
        this.name = name;
        this.ticketPrice = ticketPrice;
        this.totalSeats = totalSeats;
        this.availableSeats = availableSeats;
    }

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public double getTicketPrice() { return ticketPrice; }
    public void setTicketPrice(double ticketPrice) { this.ticketPrice = ticketPrice; }

    public int getTotalSeats() { return totalSeats; }
    public void setTotalSeats(int totalSeats) { this.totalSeats = totalSeats; }

    public int getAvailableSeats() { return availableSeats; }
    public void setAvailableSeats(int availableSeats) { this.availableSeats = availableSeats; }

    public boolean isSoldOut() { return availableSeats <= 0; }

    @Override
    public String toString() {
        return "Match{id=" + getId() + ", name='" + name + "', price=" + ticketPrice
                + ", seats=" + availableSeats + "/" + totalSeats + "}";
    }
}

File: basketball-csharp/BasketballTickets/Domain/Match.cs
namespace BasketballTickets.Domain;

public class Match : Entity<long>
{
    public string Name { get; set; } = string.Empty;
    public double TicketPrice { get; set; }
    public int TotalSeats { get; set; }
    public int AvailableSeats { get; set; }

    public bool IsSoldOut => AvailableSeats <= 0;

    public Match() { }

    public Match(long id, string name, double ticketPrice, int totalSeats, int availableSeats)
        : base(id)
    {
        Name = name;
        TicketPrice = ticketPrice;
        TotalSeats = totalSeats;
        AvailableSeats = availableSeats;
    }

    public override string ToString() =>
        $"Match{{Id={Id}, Name='{Name}', Price={TicketPrice}, Seats={AvailableSeats}/{TotalSeats}}}";
}
